provider/virtio_mem_utils.py

```diff
@@ -81,19 +96,55 @@ def check_memory_devices(device_id, cfg_req_size_param, threshold, vm, test):
             size = int(data.get("size"))
             node = int(data.get("node"))
             requested_size = int(data.get("requested-size"))
-            error_context.context("node: %d, id: %s, size: %d and requested-size: %d"
-                                  % (node, id, size, requested_size), test.log.debug)
+            error_context.context(
+                "node: %d, id: %s, size: %d and requested-size: %d"
+                % (node, id, size, requested_size),
+                test.log.debug,
+            )
             node_size = float(get_node_size(node, vm, test))
-            normalized_size = float(normalize_data_size(str(size), 'M'))
+            normalized_size = float(normalize_data_size(str(size), "M"))
             mem_limit = (node_size - normalized_size) * threshold
             if size != requested_size and normalized_size > mem_limit:
-                test.fail("Error, size: %d and requested-size: %d are not equals."
-                          % (size, requested_size))
+                test.fail(
+                    "Error, size: %d and requested-size: %d are not equals."
+                    % (size, requested_size)
+                )
             if cfg_req_size_param != "0":
-                req_size_param = int(float(normalize_data_size(cfg_req_size_param, 'B')))
+                req_size_param = int(
+                    float(normalize_data_size(cfg_req_size_param, "B"))
+                )
             else:
                 req_size_param = int(cfg_req_size_param)
             error_context.context("req_size_param: %d" % req_size_param, test.log.debug)
             if requested_size != req_size_param:
-                test.fail("Error, requested-size: %d is not the specified in the cfg: %d."
-                          % (requested_size, req_size_param))
+                test.fail(
+                    "Error, requested-size: %d is not the specified in the cfg: %d."
+                    % (requested_size, req_size_param)
+                )
+
+
+def count_memslots(vm, mem_object_id):
+    """
+    Returns the number of memslots assigned to the current virtio-mem device
+    :param vm: the VM object
+    :param mem_object_id: the ID of the memory object device
+    """
+    output = vm.monitor.info("mtree").splitlines()
+    memslots = set()
+    for memory_region in output:
+        memory_region = memory_region.strip()
+        if mem_object_id in memory_region and "memslot" in memory_region:
+            memslots.add(memory_region)
+    return len(memslots)
+
+
+def validate_memslots(expected_memslots, test, vm, mem_object_id):
```

Comment:
```suggestion
def validate_memslots(test, vm, mem_object_id, expected_memslots, timeout=10):
```
Suggest adding the timeout for handling the test case with the low-performance environment.

Reply:
What's the usage idea of the timeout? Isn't this the same as the value added to the lambda function in L149 ?

Reply:
> @YongxueHong what's the usage idea of the timeout? Isn't this the same as the value added to the lambda function in L149 ?

Hi @mcasquer 
yeah, this timeout matches the argument `timeout` of `avocado.utils.wait.wait_for` as you mentioned L149.
Sometimes, we can not get the expected slots within 10 seconds (the default) if we are running the test case in a low hardware or low-performance host, or if we are booting many VMs on the host, since the performance, we need to enlarge the timeout to make each VM can get the correct slots.
In sum, I would like to suggest that you make the timeout as configurable.

Reply:
@YongxueHong code updated and timeout now is configurable